### PR TITLE
Enhance MySQL create event parse

### PR DESCRIPTION
**Pull request by zihaoAK47**
Refer #25535.

Changes proposed in this pull request:
  - Change grammar rules
  - Add test
---

Before committing this PR, I'm sure that I have checked the following options:
- [ ✓] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ✓] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ✓] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ✓] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLDropEventStatement.java [hunk 1]
```diff
@@ -17,12 +17,11 @@
...
-import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
-import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.DDLStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.CreateEventTriggerStatement;
 import org.apache.shardingsphere.sql.parser.sql.dialect.statement.mysql.MySQLStatement;
 
 /**
  * MySQL drop event statement.
  */
-public final class MySQLDropEventStatement extends AbstractSQLStatement implements DDLStatement, MySQLStatement {
+public final class MySQLDropEventStatement extends CreateEventTriggerStatement implements MySQLStatement {
```
Comment (strongduanmu) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLDropEventStatement.java:26]: Why drop event extends CreateEventTriggerStatement?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by zihaoAK47** [comment]
@strongduanmu Completed modification please review.

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLDropEventStatement.java [hunk 2]
```diff
@@ -17,12 +17,11 @@
...
-import org.apache.shardingsphere.sql.parser.sql.common.statement.AbstractSQLStatement;
-import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.DDLStatement;
+import org.apache.shardingsphere.sql.parser.sql.common.statement.ddl.DropEventTriggerStatement;
 import org.apache.shardingsphere.sql.parser.sql.dialect.statement.mysql.MySQLStatement;
 
 /**
  * MySQL drop event statement.
  */
-public final class MySQLDropEventStatement extends AbstractSQLStatement implements DDLStatement, MySQLStatement {
+public final class MySQLDropEventStatement extends DropEventTriggerStatement implements MySQLStatement {
```
Comment (strongduanmu) [inline parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLDropEventStatement.java:26]: Why change AbstractSQLStatement to DropEventTriggerStatement? Usually only when multiple dialects contain this statement will we consider extracting the common superclass.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLDropEventStatement.java (hunk 2 above)
Comment (zihaoAK47) [reply parser/sql/statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/ddl/MySQLDropEventStatement.java]: Thank you for your guidance. I understand :blush:

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**